PhotoMap/Models/Marker.swift

```diff
@@ -5,11 +5,26 @@
 //  Created by yurykasper on 24.05.21.
 //
 
-import UIKit
+import Foundation
+import FirebaseFirestore
 
 struct Marker {
-    let image: UIImage
-    let description: String
-    let date: Date
     let category: String
+    let date: Date
+    var description: String?
+    var hashtags = [String]()
+    let photoURLString: String
+    let location: (x: Int, y: Int)
+}
+
+extension Marker {
+    init(dictionary: [String: Any]) {
+        category = dictionary["category"] as? String ?? "Default"
+        let timestamp = dictionary["date"] as? Timestamp ?? Timestamp(date: Date())
+        date = Date(timeIntervalSince1970: TimeInterval(timestamp.seconds + 54_000))
+        description = dictionary["description"] as? String
+        hashtags = dictionary["hashtags"] as? [String] ?? [String]()
+        photoURLString = dictionary["photoURL"] as? String ?? "no string-url"
+        location = dictionary["point"] as? (x: Int, y: Int) ?? (0, 0)
```

Comment:
Make it optional instead. (0, 0) means some location, it's not empty. So hypothetically it's a location.

Null Island is an imaginary island located at 0°N 0°E in the South Atlantic Ocean. This point is where the Equator meets the Prime Meridian.